lib/src/layer/polygon_layer/polygon_layer.dart

```diff
@@ -65,17 +65,19 @@ class _PolygonLayerState extends State<PolygonLayer> {
   List<_ProjectedPolygon>? _cachedProjectedPolygons;
   final _cachedSimplifiedPolygons = <int, List<_ProjectedPolygon>>{};
 
+  double? _devicePixelRatio;
+
   @override
   void didUpdateWidget(PolygonLayer oldWidget) {
     super.didUpdateWidget(oldWidget);
 
-    // Reuse cache
-    if (widget.simplificationTolerance != 0 &&
-        oldWidget.simplificationTolerance == widget.simplificationTolerance &&
-        listEquals(oldWidget.polygons, widget.polygons)) return;
-
-    _cachedSimplifiedPolygons.clear();
-    _cachedProjectedPolygons = null;
+    if (!listEquals(oldWidget.polygons, widget.polygons)) {
+      _cachedProjectedPolygons = null;
+      _cachedSimplifiedPolygons.clear();
+    } else if (!(widget.simplificationTolerance != 0 &&
+        oldWidget.simplificationTolerance == widget.simplificationTolerance)) {
+      _cachedSimplifiedPolygons.clear();
+    }
```

Comment:
Can we have some documentation here of what is the difference in both cases?

Reply:
Fixed in a4fa7c9.